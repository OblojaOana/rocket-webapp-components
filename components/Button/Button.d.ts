import { ButtonProps as BaseProps } from '@material-ui/core/Button';

export interface ButtonProps extends Omit<BaseProps, 'color' | 'size'> {
  /**
  * The color of the button.
  */
  color?: "primary" |
  "info" |
  "theme" |
  "themeNoBackground" |
  "themeWithBackground" |
  "success" |
  "warning" |
  "danger" |
  "rose" |
  "white" |
  "simple" |
  "defaultNoBackground" |
  "primaryNoBackground" |
  "infoNoBackground" |
  "successNoBackground" |
  "warningNoBackground" |
  "dangerNoBackground" |
  "roseNoBackground" |
  "white" |
  "simple" |
  "transparent"
  /**
   * If true, rounded corners are enabled.
   */
  round?: boolean
  /**
  * 	The content of the component.
  */
  children?: React.ReactNode
  /**
  * 	If true, the button will take up the full width of its container.
  */
  fullWidth?: boolean
  /**
  * If true, the button will be disabled.
  */
  disabled?: boolean
  /**
  * A custom class provided.
  */
  customClass?: string
  /**
  * If true,  the button's min width will be set to 160px.
  */
  wd?: boolean
  /**
  * If true, the button will be smaller.
  */
  justIcon?: boolean
  /**
  * If true, the button will float to the right.
  */
  right?: boolean
  /**
  * The size of the button.
  */
  size?: "sm" | "lg" | "xs"
  /**
  * The tooltip of the button.
  */
  tooltip?: string

}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/button
 *
 *   Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/Button/Button.js
 */
export default function Button(props: ButtonProps): JSX.Element;