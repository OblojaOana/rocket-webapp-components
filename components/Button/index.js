import Button from "../../core/Button";
export default Button;