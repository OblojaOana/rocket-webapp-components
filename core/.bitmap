/* THIS IS A BIT-AUTO-GENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. */

{
    "totalsoft_oss.react-mui/new/core@0.0.11": {
        "files": [
            {
                "relativePath": "src/core/index.js",
                "test": false,
                "name": "index.js"
            }
        ],
        "mainFile": "src/core/index.js",
        "trackDir": "src/core",
        "origin": "AUTHORED",
        "exported": true
    },
    "version": "14.8.8"
}