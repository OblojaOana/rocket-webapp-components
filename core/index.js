//Buttons
export { default as Button } from '@bit/totalsoft_oss.react-mui.button'
export { default as Card } from '@bit/totalsoft_oss.react-mui.card'
export { default as AddButton } from '@bit/totalsoft_oss.react-mui.add-button'
export { default as DeleteButton } from '@bit/totalsoft_oss.react-mui.delete-button'
export { default as UploadButton } from '@bit/totalsoft_oss.react-mui.upload-button'
export { default as BackToButton } from '@bit/totalsoft_oss.react-mui.back-to-button'
export { default as IconButton } from '@bit/totalsoft_oss.react-mui.icon-button'
export { default as CancelButton } from '@bit/totalsoft_oss.react-mui.cancel-button'
export { default as DownloadButton } from '@bit/totalsoft_oss.react-mui.download-button'
export { default as DownwardButton } from '@bit/totalsoft_oss.react-mui.downward-button'
export { default as EditButton } from '@bit/totalsoft_oss.react-mui.edit-button'
export { default as SaveButton } from '@bit/totalsoft_oss.react-mui.save-button'
export { default as UpwardButton } from '@bit/totalsoft_oss.react-mui.upward-button'

//Cards
export { default as IconCard } from '@bit/totalsoft_oss.react-mui.icon-card'
export { default as IconCollapseCard } from '@bit/totalsoft_oss.react-mui.icon-collapse-card'
export { default as RegularCard } from '@bit/totalsoft_oss.react-mui.regular-card'
export { default as ChartCard } from '@bit/totalsoft_oss.react-mui.chart-card'
export { default as StatsCard } from '@bit/totalsoft_oss.react-mui.stats-card'
export { default as CardTitle } from '@bit/totalsoft_oss.react-mui.card-title'
export { default as ImageCard } from '@bit/totalsoft_oss.react-mui.image-card'

//FakeText
export { default as LoadingFakeText } from '@bit/totalsoft_oss.react-mui.fake-text'

//Inputs
export { default as CustomTextField } from '@bit/totalsoft_oss.react-mui.custom-text-field'
export { default as DateTime } from '@bit/totalsoft_oss.react-mui.date-time'
export { default as DynamicInput } from '@bit/totalsoft_oss.react-mui.dynamic-input'

//NavPills
export { default as NavPills } from '@bit/totalsoft_oss.react-mui.nav-pills'

//Pagination
export { default as Pagination } from '@bit/totalsoft_oss.react-mui.pagination'

//AutoComplete  
export { default as Autocomplete } from '@bit/totalsoft_oss.react-mui.autocomplete'

//Forbidden and NotFound
export { default as Forbidden } from '@bit/totalsoft_oss.react-mui.forbidden'
export { default as NotFound } from '@bit/totalsoft_oss.react-mui.not-found'

//SimpleSlider
export { default as SimpleSlider } from '@bit/totalsoft_oss.react-mui.simple-slider'

//TimePicker
export { default as TimePicker } from '@bit/totalsoft_oss.react-mui.time-picker'

//ExpandingText
export { default as ExpandingText } from '@bit/totalsoft_oss.react-mui.expanding-text'

//Accordion
export { default as Accordion } from '@bit/totalsoft_oss.react-mui.accordion'
export { default as Piano } from '@bit/totalsoft_oss.react-mui.piano'

//Chart factory
export { default as ChartFactory } from '@bit/totalsoft_oss.react-mui.charts'

//CustomDialog
export { default as CustomDialog } from '@bit/totalsoft_oss.react-mui.custom-dialog'

//CustomLinearProgress
export { default as CustomLinearProgress } from '@bit/totalsoft_oss.react-mui.custom-linear-progress'

//Typography
export { default as Typography } from '@bit/totalsoft_oss.react-mui.typography'

//CheckInternetConnection
export { default as CheckInternetConnection } from '@bit/totalsoft_oss.react-mui.check-internet-connection'

//DialogDisplay
export { default as DialogDisplay } from '@bit/totalsoft_oss.react-mui.dialog-display'

//PasswordField
export { default as PasswordField } from '@bit/totalsoft_oss.react-mui.password-field'

//Toast
export { useToast, ToastContainer } from '@bit/totalsoft_oss.react-mui.toast'