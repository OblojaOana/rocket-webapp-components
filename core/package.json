{
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "prebuild": "rimraf build",
    "build": "npm run build:cjs &&  npm run build:copy-files",
    "build:cjs": "babel src -d build --copy-files",
    "build:copy-files": "node ./scripts/copy-files.js",
    "bit-build": "bit -v && bit import && bit build"
  },
  "devDependencies": {
    "@babel/cli": "^7.8.0",
    "@babel/core": "^7.7.7",
    "@babel/node": "^7.7.7",
    "@babel/plugin-proposal-decorators": "^7.8.3",
    "@babel/preset-env": "^7.7.7",
    "@babel/preset-react": "^7.8.0",
    "babel-eslint": "^10.0.3",
    "cross-env": "^7.0.2",
    "eslint": "^6.8.0",
    "eslint-config-prettier": "^6.7.0",
    "eslint-config-standard": "^14.1.0",
    "eslint-config-standard-react": "^9.2.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-node": "^11.0.0",
    "eslint-plugin-prettier": "^3.1.1",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-react": "^7.17.0",
    "eslint-plugin-standard": "^4.0.1",
    "gh-pages": "^2.2.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^1.19.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.1",
    "rimraf": "^3.0.0",
    "url-search-params-polyfill": "^7.0.1"
  },
  "peerDependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-flags": "^0.1.18",
    "react-router-dom": "^5.1.2",
    "react-i18next": "^10.13.2"
  },
  "sideEffects": false,
  "dependencies": {
    "@bit/totalsoft_oss.react-mui.card": "*",
    "@bit/totalsoft_oss.react-mui.not-found": "*",
    "@bit/totalsoft_oss.react-mui.button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.delete-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.upload-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.back-to-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.icon-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.cancel-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.download-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.downward-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.edit-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.save-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.upward-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.icon-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.icon-collapse-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.regular-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.chart-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.stats-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.card-title": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.image-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.fake-text": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.custom-text-field": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.date-time": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.dynamic-input": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.nav-pills": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.pagination": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.autocomplete": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.forbidden": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.simple-slider": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.time-picker": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.expanding-text": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.accordion": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.piano": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.charts": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.custom-dialog": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.custom-linear-progress": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.typography": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.check-internet-connection": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.dialog-display": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.password-field": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.toast": "^0.0.1"
  },
  "bit": {
    "env": {
      "compiler": "totalsoft_oss.react-mui/compilers/react@1.0.24"
    },
    "componentsDefaultDirectory": "components/{name}",
    "packageManager": "npm"
  }
}