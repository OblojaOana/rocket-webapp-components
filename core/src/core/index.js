// //Buttons
// export { default as Button } from './Button'
// export { default as AddButton } from './AddButton'
// export { default as DeleteButton } from './DeleteButton'
// export { default as UploadButton } from './UploadButton'
// export { default as BackToButton } from './BackToButton'
// export { default as IconButton } from './IconButton'
// export { default as CancelButton } from './CancelButton'
// export { default as DownloadButton } from './DownloadButton'
// export { default as DownwardButton } from './DownwardButton'
// export { default as EditButton } from './EditButton'
// export { default as SaveButton } from './SaveButton'
// export { default as UpwardButton } from './UpwardButton'

// //Cards
// export { default as IconCard } from './IconCard'
// export { default as IconCollapseCard } from './IconCollapseCard'
// export { default as RegularCard } from './RegularCard'
// export { default as ChartCard } from './ChartCard'
// export { default as StatsCard } from './StatsCard'
// export { default as CardTitle } from './CardTitle'
// export { default as ImageCard } from './ImageCard'

// //FakeText
// export { default as LoadingFakeText } from './FakeText'

// //Inputs
// export { default as CustomTextField } from './CustomTextField'
// export { default as DateTime } from './DateTime'
// export { default as DynamicInput } from './DynamicInput'

// //NavPills
// export { default as NavPills } from './NavPills'

// //Pagination
// export { default as Pagination } from './Pagination'

// //AutoComplete  
// export { default as Autocomplete } from './Autocomplete'

// //Forbidden and NotFound
// export { default as Forbidden } from './Forbidden/Forbidden'
// export { default as NotFound } from './NotFound/NotFound'

// //SimpleSlider
// export { default as SimpleSlider } from './SimpleSlider'

// //TimePicker
// export { default as TimePicker } from './TimePicker'

// //ExpandingText
// export { default as ExpandingText } from './ExpandingText'

// //Accordion
// export { default as Accordion } from './Accordion'
// export { default as Piano } from './Piano'

// //Chart factory
// export { default as ChartFactory } from './Charts'

// //CustomDialog
// export { default as CustomDialog } from './CustomDialog'

// //CustomLinearProgress
// export { default as CustomLinearProgress } from './CustomLinearProgress'

// //Typography
// export { default as Typography } from './Typography'

// //CheckInternetConnection
// export { default as CheckInternetConnection } from './CheckInternetConnection'

// //DialogDisplay
// export { default as DialogDisplay } from './DialogDisplay'

// //PasswordField
// export { default as PasswordField } from './PasswordField'

// //Toast
// export { useToast, ToastContainer } from './Toast'


import LoadingFakeText from '@bit/totalsoft_oss.react-mui.fake-text';
import SaveButton from '@bit/totalsoft_oss.react-mui.save-button';
import RegularCard from '@bit/totalsoft_oss.react-mui.regular-card';
import Piano from '@bit/totalsoft_oss.react-mui.piano';
import Toast from '@bit/totalsoft_oss.react-mui.toast';
import UpwardButton from '@bit/totalsoft_oss.react-mui.upward-button';
import DownwardButton from '@bit/totalsoft_oss.react-mui.downward-button';
import Forbidden from '@bit/totalsoft_oss.react-mui.forbidden';
import Card from '@bit/totalsoft_oss.react-mui.card';
import StatsCard from '@bit/totalsoft_oss.react-mui.stats-card';
import CustomDialog from '@bit/totalsoft_oss.react-mui.custom-dialog';
import UploadButton from '@bit/totalsoft_oss.react-mui.upload-button';
import Pagination from '@bit/totalsoft_oss.react-mui.pagination';
import TimePicker from '@bit/totalsoft_oss.react-mui.time-picker';
import PasswordField from '@bit/totalsoft_oss.react-mui.password-field';
import FileUploadButton from '@bit/totalsoft_oss.react-mui.file-upload-button';
import DynamicInput from '@bit/totalsoft_oss.react-mui.dynamic-input';
import Charts from '@bit/totalsoft_oss.react-mui.charts';
import NavPills from '@bit/totalsoft_oss.react-mui.nav-pills';
import DeleteButton from '@bit/totalsoft_oss.react-mui.delete-button';
import CustomLinearProgress from '@bit/totalsoft_oss.react-mui.custom-linear-progress';
import IconButton from '@bit/totalsoft_oss.react-mui.icon-button';
import CheckInternetConnection from '@bit/totalsoft_oss.react-mui.check-internet-connection';
import SimpleSlider from '@bit/totalsoft_oss.react-mui.simple-slider';
import ImageCard from '@bit/totalsoft_oss.react-mui.image-card';
import Button from '@bit/totalsoft_oss.react-mui.button';
import CustomTextField from '@bit/totalsoft_oss.react-mui.custom-text-field';
import BackToButton from '@bit/totalsoft_oss.react-mui.back-to-button';
import DateTime from '@bit/totalsoft_oss.react-mui.date-time';
import IconCollapseCard from '@bit/totalsoft_oss.react-mui.icon-collapse-card';
import DialogDisplay from '@bit/totalsoft_oss.react-mui.dialog-display';
import NotFound from '@bit/totalsoft_oss.react-mui.not-found';
import Typography from '@bit/totalsoft_oss.react-mui.typography';
import IconCard from '@bit/totalsoft_oss.react-mui.icon-card';
import EditButton from '@bit/totalsoft_oss.react-mui.edit-button';
import CardTitle from '@bit/totalsoft_oss.react-mui.card-title';
import CancelButton from '@bit/totalsoft_oss.react-mui.cancel-button';
import DownloadButton from '@bit/totalsoft_oss.react-mui.download-button';

export default { LoadingFakeText, SaveButton, RegularCard }