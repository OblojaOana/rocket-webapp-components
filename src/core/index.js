//Buttons
export { default as Button } from './Button'
export { default as AddButton } from './AddButton'
export { default as DeleteButton } from './DeleteButton'
export { default as UploadButton } from './UploadButton'
export { default as BackToButton } from './BackToButton'
export { default as IconButton } from './IconButton'
export { default as CancelButton } from './CancelButton'
export { default as DownloadButton } from './DownloadButton'
export { default as DownwardButton } from './DownwardButton'
export { default as EditButton } from './EditButton'
export { default as SaveButton } from './SaveButton'
export { default as UpwardButton } from './UpwardButton'

//Cards
export { default as IconCard } from './IconCard'
export { default as IconCollapseCard } from './IconCollapseCard'
export { default as RegularCard } from './RegularCard'
export { default as ChartCard } from './ChartCard'
export { default as StatsCard } from './StatsCard'
export { default as CardTitle } from './CardTitle'
export { default as ImageCard } from './ImageCard'

//FakeText
export { default as LoadingFakeText } from './FakeText'

//Inputs
export { default as CustomTextField } from './CustomTextField'
export { default as DateTime } from './DateTime'
export { default as DynamicInput } from './DynamicInput'

//NavPills
export { default as NavPills } from './NavPills'

//Pagination
export { default as Pagination } from './Pagination'

//AutoComplete  
export { default as Autocomplete } from './Autocomplete'

//Forbidden and NotFound
export { default as Forbidden } from './Forbidden/Forbidden'
export { default as NotFound } from './NotFound/NotFound'

//SimpleSlider
export { default as SimpleSlider } from './SimpleSlider'

//TimePicker
export { default as TimePicker } from './TimePicker'

//ExpandingText
export { default as ExpandingText } from './ExpandingText'

//Accordion
export { default as Accordion } from './Accordion'
export { default as Piano } from './Piano'

//Chart factory
export { default as ChartFactory } from './Charts'

export { default as CustomDialog } from './CustomDialog'

export { default as CustomLinearProgress } from './CustomLinearProgress'

export { default as Typography } from './Typography'

export { default as CheckInternetConnection } from './CheckInternetConnection'

export { default as DialogDisplay } from './DialogDisplay'

export { default as PasswordField } from './PasswordField'

//Toast
export { useToast, ToastContainer } from './Toast'