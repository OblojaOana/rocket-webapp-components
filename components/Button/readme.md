##### Copy this snapshot in the overview panel
 ```js
import React from 'react';
import Button from '@bit/totalsoft_oss.react-mui.button';
import { makeStyles } from '@material-ui/core/styles';
import DeleteIcon from '@material-ui/icons/Delete';
import CloudUploadIcon from '@material-ui/icons/CloudUpload';
import KeyboardVoiceIcon from '@material-ui/icons/KeyboardVoice';
import Icon from '@material-ui/core/Icon';
import SaveIcon from '@material-ui/icons/Save';
import SendIcon from '@material-ui/icons/Send';

const useStyles = makeStyles((theme) => ({
	button: {
		margin: theme.spacing(5),
	},
}));

const App = () => {
	const classes = useStyles();

	return (
		<div>
			<Button
				color="warning"
				className={classes.button}
				startIcon={<DeleteIcon />}
				size="sm"
			>
				Delete
      </Button>
			<Button
				color="primary"
				className={classes.button}
				endIcon={<SendIcon />}
				size="sm"
			>
				Send
      </Button>
			<Button
				color="info"
				className={classes.button}
				startIcon={<CloudUploadIcon />}
				size="sm"
			>
				Upload
      </Button>
			<Button
				color="danger"
				className={classes.button}
				startIcon={<KeyboardVoiceIcon />}
				size="sm"
			>
				Talk
      </Button>
			<Button
				color="info"
				size="small"
				className={classes.button}
				startIcon={<SaveIcon />}
				size="sm"
			>
				Save
      </Button>
			<Button
				color="success"
				size="large"
				className={classes.button}
				startIcon={<SaveIcon />}
				size="sm"
			>
				Save
      </Button>
		</div>
	)
}

export default (
	<App />
)
 ```



 ##### Copy this snapshot in the overview panel
 ```js
import React from 'react';
import Button from '@bit/totalsoft_oss.react-mui.button';
import { makeStyles } from '@material-ui/core/styles';
import DeleteIcon from '@material-ui/icons/Delete';
import CloudUploadIcon from '@material-ui/icons/CloudUpload';
import KeyboardVoiceIcon from '@material-ui/icons/KeyboardVoice';
import Icon from '@material-ui/core/Icon';
import SaveIcon from '@material-ui/icons/Save';
import SendIcon from '@material-ui/icons/Send';

const useStyles = makeStyles((theme) => ({
	margin: {
		margin: theme.spacing(1),
	},
	extendedIcon: {
		marginRight: theme.spacing(1),
	},
}));

const App = () => {
	const classes = useStyles();

	return (
		<div>
			<Button size="xs" color="info" className={classes.margin}>
				Small
        </Button>
			<Button size="sm" color="info" className={classes.margin}>
				Medium
        </Button>
			<Button size="lg" color="info" className={classes.margin}>
				Large
        </Button>
		</div>
	)
}

export default (
	<App />
)
 ```