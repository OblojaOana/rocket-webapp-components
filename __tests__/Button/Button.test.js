import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button/Button';
import { Button as ButtonBase } from "@material-ui/core";

describe('Button', () => {
    it('snapshot - warning', () => {
        const component = renderer.create(<Button color="warning">
            Warning
        </Button>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - primary', () => {
        const component = renderer.create(<Button color="primary">
            primary
        </Button>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - info', () => {
        const component = renderer.create(<Button color="info">
            info
        </Button>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - success', () => {
        const component = renderer.create(<Button color="success">
            success
        </Button>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - danger', () => {
        const component = renderer.create(<Button color="danger">
            danger
        </Button>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - rose', () => {
        const component = renderer.create(<Button color="rose">
            rose
        </Button>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('button will be rounded', () => {
        const wrapper = renderer.create(<Button round={true} color="rose">
            Warning
        </Button>);

        const testInstance = wrapper.root;

        expect(testInstance.findByType(ButtonBase).props.className).toContain('round');
    });


    it('button will not be rounded', () => {
        const wrapper = renderer.create(<Button round={false} color="rose">
            Warning
        </Button>);

        const testInstance = wrapper.root;

        expect(testInstance.findByType(ButtonBase).props.className).not.toContain('round');
    });

    it('button will not be fullWidth', () => {
        const wrapper = renderer.create(<Button fullWidth={false} color="rose">
            Warning
        </Button>);

        const testInstance = wrapper.root;

        expect(testInstance.findByType(ButtonBase).props.className).not.toContain('fullWidth');
    });

    it('button will be fullWidth', () => {
        const wrapper = renderer.create(<Button fullWidth={true} color="rose">
            Warning
        </Button>);

        const testInstance = wrapper.root;

        expect(testInstance.findByType(ButtonBase).props.className).toContain('fullWidth');
    });

    it('button will be disabled', () => {
        const wrapper = renderer.create(<Button disabled={true} color="rose">
            Warning
        </Button>);

        const testInstance = wrapper.root;

        expect(testInstance.findByType(ButtonBase).props.className).toContain('disabled');
    });


    it('button will not be disabled', () => {
        const wrapper = renderer.create(<Button disabled={false} color="rose">
            Warning
        </Button>);

        const testInstance = wrapper.root;

        expect(testInstance.findByType(ButtonBase).props.className).not.toContain('disabled');
    });
});