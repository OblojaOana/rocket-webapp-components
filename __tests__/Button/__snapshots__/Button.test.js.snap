// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot - danger 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 makeStyles-danger-8"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    danger
  </span>
</button>
`;

exports[`Button snapshot - info 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 makeStyles-info-5"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    info
  </span>
</button>
`;

exports[`Button snapshot - primary 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 makeStyles-primary-4"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    primary
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Button snapshot - rose 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 makeStyles-rose-9"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    rose
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Button snapshot - success 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 makeStyles-success-6"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    success
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Button snapshot - warning 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 makeStyles-warning-7"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Warning
  </span>
</button>
`;
